src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ImportCompletionItem.cs

```diff
@@ -71,63 +88,58 @@ public static CompletionItem CreateAttributeItemWithoutSuffix(CompletionItem att
-                var symbol = compilation.GetTypeByMetadataName(metadataName);
-                if (symbol != null)
-                {
-                    var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-
-                    // We choose not to display the number of "type overloads" for simplicity. 
-                    // Otherwise, we need additional logic to track internal and public visible
-                    // types separately, and cache both completion items.
-                    return await CommonCompletionUtilities.CreateDescriptionAsync(
-                        document.Project.Solution.Workspace,
-                        semanticModel,
-                        position: 0,
-                        symbol,
-                        overloadCount: 0,
-                        supportedPlatforms: null,
-                        cancellationToken).ConfigureAwait(false);
-                }
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+                // We choose not to display the number of "type overloads" for simplicity. 
+                // Otherwise, we need additional logic to track internal and public visible
+                // types separately, and cache both completion items.
+                return await CommonCompletionUtilities.CreateDescriptionAsync(
+                    document.Project.Solution.Workspace,
+                    semanticModel,
+                    position: 0,
+                    symbol,
+                    overloadCount: 0,
+                    supportedPlatforms: null,
+                    cancellationToken).ConfigureAwait(false);
             }
 
             return CompletionDescription.Empty;
         }
 
-        private static string GetAritySuffix(int arity)
-        {
-            Debug.Assert(arity > 0);
-            return (arity <= s_aritySuffixesOneToNine.Length)
-                ? s_aritySuffixesOneToNine[arity - 1]
-                : string.Concat(GenericTypeNameManglingString, arity.ToString(CultureInfo.InvariantCulture));
-        }
-
         private static string GetFullyQualifiedName(string namespaceName, string typeName)
             => namespaceName.Length == 0 ? typeName : namespaceName + "." + typeName;
 
-        private static string GetMetadataName(CompletionItem item)
+        private static ISymbol? GetSymbol(CompletionItem item, Compilation compilation)
         {
+            // If we have SymbolKey data (i.e. this is an extension method item), use it to recover symbol
+            if (item.Properties.TryGetValue(SymbolKeyData, out var symbolId))
+            {
+                return SymbolKey.ResolveString(symbolId, compilation).GetAnySymbol();
+            }
+
+            // Otherwise, this is a type item, so we don't have SymbolKey data. But we should still have all
+            // the data to construct its full metadata name
```

Comment:
dumb question.  why not just have the TypeItem do the same thing the extension method item does?  then you can operate on both consistently.

Reply:
Because after the cached items are created, type items don't operate on symbols at all (except for getting description), holding on to those SymbolKey data in cache seems wasteful. Whereas, we need symbols to provide extension method items every time.